nestkernel/per_thread_bool_indicator.cpp

```diff
@@ -0,0 +1,137 @@
+/*
+ *  per_thread_bool_indicator.cpp
+ *
+ *  This file is part of NEST.
+ *
+ *  Copyright (C) 2004 The NEST Initiative
+ *
+ *  NEST is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  NEST is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "per_thread_bool_indicator.h"
+
+namespace nest
+{
+
+BoolIndicatorUInt64::BoolIndicatorUInt64()
+  : status_( false_uint64 )
+{
+}
+
+BoolIndicatorUInt64::BoolIndicatorUInt64( const bool status )
+  : status_( status )
+{
+}
+
+bool
+BoolIndicatorUInt64::is_true() const
+{
+  return ( status_ == true_uint64 );
+}
+
+bool
+BoolIndicatorUInt64::is_false() const
+{
+  return ( status_ == false_uint64 );
+}
+
+void
+BoolIndicatorUInt64::set_true()
+{
+  status_ = true_uint64;
+}
+
+void
+BoolIndicatorUInt64::set_false()
+{
+  status_ = false_uint64;
+}
+
+void
+BoolIndicatorUInt64::logical_and( const bool status )
+{
+  status_ = ( static_cast< bool >( status_ ) and status );
```

Comment:
The right hand side results in a `bool`, which is then converted to either integer `0` or `1` when assigned to `status_`. This is consistent with the initial assignment of `true` and `false` to the constants `true_uint64` and `false_uint64`, respectively. We could use an `if/else` here to assign `true_uint64/false_uint64` but this might be a bit over the top.